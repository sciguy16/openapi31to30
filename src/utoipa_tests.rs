use crate::{convert, test::progenitor_test};
use serde::{Deserialize, Serialize};
use utoipa::openapi::OpenApi;
use utoipa_axum::{router::OpenApiRouter, routes};

mod nullable;
mod path;
mod verbs;

#[derive(utoipa::ToSchema, Deserialize, Serialize)]
struct User {
    id: i32,
}

#[derive(utoipa::ToSchema, Deserialize, Serialize)]
struct UserWithNullableField {
    id: i32,
    name: Option<String>,
}

#[test]
fn basic_get() {
    let (_router, api): (axum::Router, OpenApi) = OpenApiRouter::new()
        .routes(routes!(verbs::get_user))
        .split_for_parts();
    let yaml = serde_yaml::to_string(&api).unwrap();
    insta::assert_snapshot!("basic-before", &yaml);
    let converted = convert(&yaml).unwrap();
    insta::assert_snapshot!("basic-converted", &converted);
    progenitor_test(&converted);
}

#[test]
fn basic_verbs() {
    let (_router, api): (axum::Router, OpenApi) = OpenApiRouter::new()
        .routes(routes!(
            verbs::get_user,
            verbs::put_user,
            verbs::post_user,
            verbs::patch_user,
            verbs::delete_user
        ))
        .split_for_parts();
    let yaml = serde_yaml::to_string(&api).unwrap();
    insta::assert_snapshot!("verbs-before", &yaml);
    let converted = convert(&yaml).unwrap();
    insta::assert_snapshot!("verbs-converted", &converted);
    progenitor_test(&converted);
}

#[test]
fn paths() {
    let (_router, api): (axum::Router, OpenApi) = OpenApiRouter::new()
        .routes(routes!(
            path::get_user,
            path::put_user,
            path::post_user,
            path::patch_user,
            path::delete_user
        ))
        .split_for_parts();
    let yaml = serde_yaml::to_string(&api).unwrap();
    insta::assert_snapshot!("paths-before", &yaml);
    let converted = convert(&yaml).unwrap();
    insta::assert_snapshot!("paths-converted", &converted);
    progenitor_test(&converted);
}

#[test]
fn nullable() {
    let (_router, api): (axum::Router, OpenApi) = OpenApiRouter::new()
        .routes(routes!(nullable::put_user, nullable::post_user))
        .split_for_parts();
    let yaml = serde_yaml::to_string(&api).unwrap();
    insta::assert_snapshot!("nullable-before", &yaml);
    let converted = convert(&yaml).unwrap();
    insta::assert_snapshot!("nullable-converted", &converted);
    progenitor_test(&converted);
}
