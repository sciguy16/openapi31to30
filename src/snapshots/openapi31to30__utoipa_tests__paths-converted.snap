---
source: src/utoipa_tests.rs
expression: "&converted"
---
openapi: 3.0.3
info:
  title: utoipa-axum
  description: Utoipa's axum bindings for seamless integration for the two
  contact:
    name: Juha Kukkonen
    email: juha7kukkonen@gmail.com
  license:
    name: MIT OR Apache-2.0
    identifier: MIT OR Apache-2.0
  version: 0.2.0
paths:
  /user/{id}:
    get:
      operationId: get_user
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
          format: int64
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
    put:
      operationId: put_user
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
          format: int64
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/User'
        required: true
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
    post:
      operationId: post_user
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
          format: int64
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/User'
        required: true
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
    delete:
      operationId: delete_user
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
          format: int64
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
    patch:
      operationId: patch_user
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
          format: int64
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/User'
components:
  schemas:
    User:
      type: object
      required:
      - id
      properties:
        id:
          type: integer
          format: int32
